advanced/dapps/react-dapp-v2/src/helpers/tezos.ts

```diff
@@ -0,0 +1,27 @@
+import { TezosToolkit } from '@taquito/taquito';
+import { TezosChainData } from "../chains/tezos";
+
+export async function apiGetTezosAccountBalance(
+    address: string,
+    networkId: string
+  ) {
+    
+    // check if networkId is in the list of TezosChainData
+    if (!TezosChainData[networkId]) { 
+        throw new Error(`Unsupported networkId: ${networkId}`);
+    }
+    const rpc = TezosChainData[networkId].rpc[0];
+
+    const Tezos = new TezosToolkit(rpc);
```

Comment:
creating a new toolkit object for every call of getBalance does not make sense. Better to initialize this tezostoolkit outside in another function, and reuse it here. Similarly for rpc and chain data. get balance should simply call the getBalance method. You can extract the setup in another function and call that setup function at the beginning during initialization.